superset/datasets/metrics/commands/delete.py

```diff
@@ -40,13 +38,13 @@ def __init__(self, dataset_id: int, model_id: int):
         self._model_id = model_id
         self._model: Optional[SqlMetric] = None
 
-    def run(self) -> Model:
+    def run(self) -> None:
         self.validate()
+        assert self._model
+
         try:
-            if not self._model:
```

Comment:
DRY. The `validate` method already raises a `DatasetMetricNotFoundError` if the metric is not found.